src/test/java/com/github/commonsrdf/api/AbstractRDFTermFactoryTest.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.commonsrdf.api;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class AbstractRDFTermFactoryTest {
+
+	private RDFTermFactory factory;
+
+	@Test
+	public void createBlankNode() throws Exception {
+		BlankNode bnode = factory.createBlankNode();
+		String ntriplesString = bnode.ntriplesString();
+		assertTrue("ntriples must start with _:",
+				ntriplesString.startsWith("_:"));
+		assertTrue("Internal identifier can't be empty", bnode
+				.internalIdentifier().length() > 0);
+		assertEquals("ntriples does not correspond with internal identifier",
```

Comment:
The internal identifiers that may be used are not limited to the N-Triples subset, but that may not be clear in the current contract. Hence, the point I was trying to make was that this test, which is being designed as a compliance test for all commons-rdf implementations, should not assume that the internal identifiers are either preserved or limited to the ntriples subset.

As you say, it would probably be better to either remove ntriplesString or make it an optional method via the unsupported operation route for this reason.